### DISCUSSION TITLE
make: *** No rule to make target,needed by `bin/embedded.o'

### DISCUSSION THREAD

**Discussion by uranus829** [ipxe/ipxe#936] [General]
Hello,
make bin/undionly.kpxe EMBED=/tftpboot/default.ipxe
make: *** No rule to make target `/tftpboot/default.ipxe', needed by `bin/embedded.o'.  Stop.

make bin-x86_64-efi/ipxe.efi  EMBED=/tftpboot/dafault.ipxe is no problem.

When compiling, it prompts that bin/embedded.o cannot be found. How to solve it? Thanks

**Comment by stappersg** [comment]
> How to solve it?

Note the comma in
```text
make: *** No rule to make target `/tftpboot/default.ipxe', needed by `bin/embedded.o'.  Stop.
```

So see that the actual error is
```text
No rule to make target `/tftpboot/default.ipxe'
```

**Comment by NiKiZe** [comment]
Make sure the file actually exists. Verify that the filename that is used is the actual filename, and not something that just looks like the same. Try an ls on filename (copy paste to verify same chars)

**Comment by mcb30** [comment]
> Hello, make bin/undionly.kpxe EMBED=/tftpboot/default.ipxe make: *** No rule to make target `/tftpboot/default.ipxe', needed by `bin/embedded.o'. Stop.
> 
> `make bin-x86_64-efi/ipxe.efi EMBED=/tftpboot/dafault.ipxe` is no problem.

It looks as though you have a typo in your filename: `default.ipxe` vs `dafault.ipxe`

**Reply by uranus829** [reply]
It has been solved, thank you, this binutils dependency component is missing.
